polyfactory/factories/base.py

```diff
@@ -103,6 +94,17 @@
 F = TypeVar("F", bound="BaseFactory[Any]")
 
 
+class BuildContext(TypedDict):
+    seen_models: set[type]
+
+
+def _get_build_context(build_context: BuildContext | None) -> BuildContext:
+    if build_context is None:
+        return {"seen_models": set()}
+
+    return copy.deepcopy(build_context)
```

Comment:
Deepcopy is used here to avoid mutation of the parent build context in child contexts.